### PR TITLE
fix: Upload item removed should keep status className

### LINKED ISSUE #36286: Upload Error状态删除时有蓝色Icon闪过
### Reproduction link

[https://ant.design/components/upload-cn/#components-upload-demo-picture-card](https://ant.design/components/upload-cn/#components-upload-demo-picture-card)

### Steps to reproduce

1、这是官网文档中照片墙的例子
2、删除那张代表Error状态的图片
3、在删除瞬间可以看到有个蓝色的图片ICON闪过

### What is expected?

没有蓝色的图片ICON闪过

### What is actually happening?

有蓝色的图片ICON闪过

| Environment | Info |
| --- | --- |
| antd | 4.21.4 |
| React | 17.0.2 |
| System | macOS Catalina 10.15.7 |
| Browser | Chrome 103.0.5060.53 |

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #36286

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fixed Upload removing file status color is wrong.          |
| 🇨🇳 Chinese | 修复 Upload 移除文件时状态色会变化的问题。           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/upload/Upload.tsx [hunk 1]
```diff
@@ -262,15 +262,8 @@ const InternalUpload: React.ForwardRefRenderFunction<unknown, UploadProps> = (pr
       const removedFileList = removeFileItem(file, mergedFileList);
 
       if (removedFileList) {
-        currentFile = { ...file, status: 'removed' };
-        mergedFileList?.forEach(item => {
-          const matchKey = currentFile.uid !== undefined ? 'uid' : 'name';
-          if (item[matchKey] === currentFile[matchKey] && !Object.isFrozen(item)) {
-            item.status = 'removed';
-          }
-        });
```
Comment (afc163) [inline components/upload/Upload.tsx:271]: 这个 status removed 没啥用。

**Comment by daxiazilong** [comment]
image
直接dao导致我项目bug

**Comment by daxiazilong** [comment]
> @daxiazilong 方便提供一个重现么？

filelist  维护成一个list 然后你往里塞数据  再点击删除小图标的时候 之前版本他那个 file 的status应该为remove 但现在好像啥也没了

**Comment by thangtran2307** [comment]
Hi, this causing the bug in my project. Previously I was checking the file status to update the state then now you remove the `removed` status

image

File: components/upload/Upload.tsx (hunk 1 above)
Comment (zombieJ) [reply components/upload/Upload.tsx:271]: 这个移除后，删除时触发事件拿到的就是前一个状态了：

**Pull request merged**